docs/presentation/RepeatableRelationshipPathSpecification.md

```diff
@@ -0,0 +1,108 @@
+
+### Attribute: `targetClass`
+
+> **Default value:** Target ECClass of the [relationship](#attribute-relationship) if the [direction](#attribute-direction) is `"Forward"` or
+> source ECClass if the [direction](#attribute-direction) is `"Backward"`.
+
+This attribute may be used to specialize the target of the relationship. E.g. when relationship points to a class like `bis.Element`, this
+attribute allows specializing it to `bis.PhysicalElement` or some other `bis.Element` subclass.
+
+### Attribute: `count`
+
+> **Default value:** `1`
+
+This attribute specifies the number of times the relationship should be traversed.
+
+The special `"*"` value makes the step recursive, which means the relationship is traversed as long as new instances are found by
+using output of the previous step as input for the current step.
+
+Example:
+
+```JSON
+{
+  "relationship": { "schemaName": "MySchema", "className": "MyRelationship" },
+  "direction": "Forward",
+  "count": "*"
+}
+```
+
+Results of the above specification would consist of the following:
+
+```Text
+Outputs = [
+  Source -> MySchema.MyRelationship -> Targets
+  Source -> MySchema.MyRelationship -> MySchema.MyRelationship -> Targets
+  ...
+  Source -> MySchema.MyRelationship -> ... -> MySchema.MyRelationship -> Targets
+]
+```
+
+## Examples
+
+When the [`count` attribute](#attribute-count) is omitted or set to `1`, the specification works similarly to [RelationshipPathSpecification](./RelationshipPathSpecification.md). See its [examples section](./RelationshipPathSpecification.md#examples) for those simpler cases.
+
+### Jumping through the same relationship multiple times
+
+```ts
+[[include:RepeatableRelationshipPathSpecification.SingleStepWithCount.Ruleset]]
+```
+
+![Content of the grand-parent element](./media/repeatablerelationshippathspecification-singlestep-with-count.png)
+
+### Jumping through the relationship recursively
+
+```ts
+[[include:RepeatableRelationshipPathSpecification.RecursiveSingleStep.Ruleset]]
+```
+
+When the root subject is provided as input, content for all its child elements is returned:
+
+![Content of all root subject's child elements](./media/repeatablerelationshippathspecification-recursivesinglestep.png)
+
+### Combining recursive and non-recursive steps
+
+```ts
+[[include:RepeatableRelationshipPathSpecification.RecursiveAndNonRecursiveSpecificationsCombination.Ruleset]]
+```
+
+When a physical model is provided as input, categories' content of all its elements and their children is returned:
+
+![Categories' content of model elements and their children](./media/repeatablerelationshippathspecification-combinedsteps.png)
+
+### Combining multiple recursive steps
+
+```ts
+[[include:RepeatableRelationshipPathSpecification.MultipleRecursiveSpecificationsCombination.Ruleset]]
+```
+
+![Content of multiple recursive relationship steps](./media/repeatablerelationshippathspecification-combinedrecursivesteps.png)
```

Comment:
I'm asking this because this page doesn't explain or I perhaps missed it. Why the result of the last example includes instances of intermediate path steps while all other examples don't?

Reply:
Because `count: "*"` means that the step is optional (we're making the step 0+ times). Which means we're taking results of the previous step + any instances we find making the step recursively.

Reply:
Could you please extend `count` attribute section with another example where the relationship path consists of two specifications with `"*"` count?

Reply:
Updated, please have a look

Reply:
I see that this pattern gets complicated really quickly, so I don't think we can cover everything with just one example.

Maybe `count` attribute documentation should only say that the special value means an unbounded number of recursive traversals (including zero), and describe the logic behind recursive traversal in another section. 

The examples section could be renamed to "Recursive traversal rules" or similar. The existing examples cover some, but not all cases:

* Fixed number of recursive traversals (covered)
* Unbounded number of recursive traversals with `"*"` count (covered)
* Combination of the above two cases (covered)
* Two unbounded traversals, first of which can be skipped given the right class and relationship structure
* Two unbounded traversals, second of which can be skipped given the right class and relationship structure
* Case where two different instance types are returned (covered)

Reply:
> Two unbounded traversals, first of which can be skipped given the right class and relationship structure (can this even happen and does it make any sense?)

Yes, this can happen. An example could be `ElementOwnsChildElements + ElemenRefersToElements` where the input is a parent element. The recursive step of `ElementOwnsChildElements` may return nothing and in that case only the given parent element would be used as input to the `ElemenRefersToElements` step.

Problem is that our test iModel is not suitable to reproduce this kind of case, so I can't provide a snippet that can be tested. I updated the description of the `count: "*"` meaning - hopefully that will be sufficient. Please have a look.

Reply:
It is better than before, thanks